ext/MitsubaLoader/CMitsubaLoader.cpp

```diff
@@ -910,13 +910,14 @@ void CMitsubaLoader::genBSDFtreeTraversal(SContext& ctx, const CElementBSDF* bsd
 		bool visited;
 	};
 	core::stack<stack_el> stack;
+	uint32_t firstFreeNormalReg = 1u;//normal reg val 0 means geom normal without any perturbations
```

Comment:
you can use ordinary registers for normals (vec3 is a vec3 after all)

Reply:
one more note, it would be ideal if the top level BSDF always wrote its result to Register 0